<launch>

  <arg name="world_path" default="$(find uavros_gazebo)/worlds/precision_landing.world" />
  <arg name="sdf_path" default="$(find uavros_gazebo)/models/iris_depth_camera_down/iris_depth_camera_down.sdf" />

  <include file="$(find uavros_gazebo)/launch/camera_down_px4.launch">
    <arg name="world" value="$(arg world_path)" />
    <arg name="sdf" value="$(arg sdf_path)" />   
  </include>
    
  <include file="$(find uavros_kcffollow_simulation)/launch/kcf_detection_node.launch" >
  </include>

  <include file="$(find uavros_kcffollow_simulation)/launch/kcf_tracking_pid.launch" >
 	<arg name="search_alt_" value="7" />
 	<arg name="markers_id_" value="0" /> <!-- 0 -->
 	<arg name="desire_pose_x" value="0" />
 	<arg name="desire_pose_y" value="0.052" /> <!--0.052--> <!-- y方向估计不准，多了0.052m -->
 	<arg name="desire_pose_z" value="0.2" />  <!-- z方向估计不准 --> <!--1.2-->
 	<arg name="desire_yaw_" value="0" /> <!-- 0 -->
 	<arg name="PidXY_p" value="2.1" /> <!-- 2.1 -->
 	<arg name="PidXY_d" value="1.5" /> <!-- 1.5 -->
 	<arg name="PidXY_i" value="0" /> <!-- 0.1 --> <!-- 有积分的话一会儿就到很大的速度了 -->   
 	<arg name="PidZ_p" value="0.1" /> <!-- 0.1 -->
 	<arg name="PidZ_d" value="0.05" /> <!-- 0.05 -->
 	<arg name="PidZ_i" value="0" /> <!-- 0 -->
 	<arg name="PidYaw_p" value="0.1" /> <!-- 0.1 -->
 	<arg name="PidYaw_d" value="0" /> <!-- 0 -->
 	<arg name="PidYaw_i" value="0" /> <!-- 0 -->
	<arg name="Thres_count_detect" default="30" /> <!-- 30 -->
	<arg name="Thres_count_approach" default="30" /> <!-- 40 -->
	<arg name="Thres_count_small" default="40" /> <!-- 12 --> 
	<arg name="landing_vel" default="-0.08" /> <!-- -0.22 -->

	<param name="KCFv_p" value = "3"/><!-- 1.2 -->
	<param name="KCFv_d" value = "0"/>
	<param name="KCFv_i" value = "0"/>
	<param name="KCFw_p" value = "1"/><!-- 0.5 -->
	<param name="KCFw_d" value = "0"/>
	<param name="KCFw_i" value = "0"/>
	<param name="desireAngle_x" value = "0"/><!--rad -->
	<param name="desireAngle_y" value = "0"/><!-- rad -->

  </include>

    <node pkg="uavros_gazebo" type="keyboard_control_px4.py" name="keyboard_control_px4" output="screen" launch-prefix="gnome-terminal --tab -e">
    </node>

    <node pkg="rqt_image_view" type="rqt_image_view" name="rqt_image_view" output="screen" launch-prefix="gnome-terminal --tab -e">
    </node>


</launch>




	

